src/common/certificate-authorities/inject-system-cas.injectable.ts

```diff
@@ -22,15 +22,30 @@ const injectSystemCAsInjectable = getInjectable({
     const requestSystemCAs = di.inject(requestSystemCAsInjectionToken);
 
     return async () => {
-      for (const cert of await requestSystemCAs()) {
-        if (isCertActive(cert)) {
-          if (Array.isArray(globalAgent.options.ca) && !globalAgent.options.ca.includes(cert)) {
-            globalAgent.options.ca.push(cert);
-          } else {
-            globalAgent.options.ca = [cert];
-          }
+      const certs = await requestSystemCAs();
+
+      if (certs.length === 0) {
+        // Leave the global option alone
+        return;
+      }
+
+      const cas = Array.isArray(globalAgent.options.ca)
+        ? globalAgent.options.ca
+        : globalAgent.options.ca
+          ? [globalAgent.options.ca]
+          : [];
```

Comment:
Are you sure this does what you want? I never remember the precedence rule for `? :` used with itself and I've seen this construct used incorrectly before, causing a bug. Why not just write it more explicitly?

Reply:
Yes I am sure that this is correct but I can rewrite it if you want.